src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -167,6 +167,29 @@ public SnapshotService(
     this.snapshotBuilderSettingsDao = snapshotBuilderSettingsDao;
   }
 
+  public String generateUpdatedSnapshotNameIfByRequest(SnapshotRequestModel model) {
```

Comment:
To me, this method name is too specific about its implementation. All the caller needs to know that the method returns the requested snapshot name, which could be generated in some cases. If more cases are added, we'd want to avoid updating the method name, as its behavior relative to the caller wouldn't change.
```suggestion
  public String getSnapshotName(SnapshotRequestModel model) {
```